### PR TITLE
switch http public interface to use relay http server

### PR DESCRIPTION
**Pull request by willscott**

### CODE REVIEW

**Comment by willscott** [comment]
(I didn't work through the circular dependency introduced here. @nikkolasg part of this is the lack of understanding of which package is circular in a way that it shouldn't be.

My current thought is:
the utilities in `beacon` (`NextRound` / etc) shouldn't be there, so that `http` and the tests don't need to depend on the full beacon package?

File: http/server.go [hunk 1]
```diff
@@ -87,9 +86,9 @@ func (h *handler) PublicRand(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	roundExpectedTime := beacon.TimeOfRound(h.groupInfo.Period, h.groupInfo.GenesisTime, roundN)
+	roundExpectedTime := time.Unix(h.groupInfo.GenesisTime, 0).Add(h.groupInfo.Period * time.Duration(roundN))
```
Comment (nikkolasg) [inline http/server.go:89]: Why not keeping the beacon method ? If possible I would prefer to use a single point to compute the time - round mappings -
For example, here round 1 is set after one period after the genesis no ? While it is not the case, the first round happens at genesis time (round 0 is the genesis block)

Comment (willscott) [reply http/server.go:89]: was avoiding circular dependencies:
beacon depends on net already, so this keeps a lot of code from being able to use this handler if it depends on that.

Comment (nikkolasg) [reply http/server.go:89]: I'm not sure I see the circular dependency here - Ive changed that line locally to using the  beacon package and it works. 
Also, my first comment still holds out, I don't believe you return the correct timestamp - hence I would insist to use the same method :D 

I am totally fine to move these functions (`TimeOfRound` `NextRound` etc ) into a separate package if that is a trouble !

**Pull request merged**